client/src/pages/CreateWorkout/components/ExerciseAccordion/ExerciseAccordion.jsx

```diff
@@ -21,19 +22,19 @@ const ExerciseAccordion = ({
 
   const handleEditClick = () => {
     console.log("edit exercise requested ...");
-    handleEditExercise(exercise);
+    handleEditExercise(exercise, exerciseNumber);
   };
 
   const handleDeleteClick = () => {
     console.log("delete exercise requested ...");
-    handleDeleteExercise(exercise.id);
+    handleDeleteExercise(exerciseNumber);
   };
 
   return (
     <>
       <div className=" w-11/12 rounded-lg border-2 border-solid border-black text-center sm:mt-0 sm:w-4/5">
         <div className=" m-auto flex items-center justify-between p-3">
-          <h3 className="my-0 w-12 basis-auto font-bold ">{exercise.id}</h3>
+          <h3 className="my-0 w-12 basis-auto font-bold ">{exerciseNumber}</h3>
```

Comment:
Awesome! Now, if you want to simplify this even further - you could get rid of the concept of `exerciseNumber` where `exerciseNumber = index + 1` and just pass around the actual `exerciseIndex` everywhere....

and then the only place that you'd have to do +1 is where you are actually displaying it to the user, i.e. right here :)